components/server/src/workspace/gitpod-server-impl.ts

```diff
@@ -1707,16 +1795,31 @@ export class GitpodServerImpl implements GitpodServerWithTracing, Disposable {
     }
 
     public async hasPermission(ctx: TraceContext, permission: PermissionName): Promise<boolean> {
+        traceAPIParams(ctx, { permission });
+
         const user = this.checkUser("hasPermission");
         return this.authorizationService.hasPermission(user, permission);
     }
 
     preparePluginUpload(ctx: TraceContext, params: PreparePluginUploadParams): Promise<string> {
+        traceAPIParams(ctx, { params });
+
         const user = this.checkUser("preparePluginUpload");
         return this.pluginService.preparePluginUpload(params, user.id);
     }
 
     async resolvePlugins(ctx: TraceContext, workspaceId: string, params: ResolvePluginsParams): Promise<ResolvedPlugins> {
+        traceAPIParams(ctx, {
+            workspaceId,
+            params: {
+                ...censor(params, "config"),
+                config: {
+                    vscode: params?.config?.vscode,
+                },
```

Comment:
**Nit**: Cool trick censoring `config` then partially adding it back!

However, I think you're changing the semantics here (e.g. if `config` was `undefined`, in tracing it suddenly becomes defined as an object like `config = { vscode: undefined }`.

Maybe safer like this:

```suggestion
                ...censor(params, "config"),
                ...(!!params.config ? : {
                    config: {
                        vscode: params.config.vscode,
                    },
                } : {}),
```

Reply:
Yep. Went with conciseness over preciseness here... but now that you found it: here you go! :wink: